lib/http/2/stream.rb

```diff
@@ -78,6 +78,7 @@ def initialize(connection:, id:, weight: 16, dependency: 0, exclusive: false, pa
       @dependency = dependency
       process_priority(weight: weight, stream_dependency: dependency, exclusive: exclusive)
       @local_window  = connection.local_settings[:settings_initial_window_size]
+      @recv_window_size = 0
```

Comment:
Why do you need this? Aren't you simply counting the exact same thing twice, in one case by decrementing from @local_window and in another by incrementing @recv_window_size?

Reply:
most of the logic of the code snippet is directly inherited from nghttp, which carries the recv_window_size around. More info about usefulness in the other comments.

Reply:
you might be right though, as nghttp doesn't seem to mess with their local_window_size parameters, but [calculate the local window as another operation dependent from the recv_window_size](https://github.com/nghttp2/nghttp2/blob/2bf3680d870953010d7e1e6e4a66510f8458cc3c/lib/nghttp2_session.c#L7113-L7115).